pulsar-client-cpp/include/pulsar/Authentication.h

```diff
@@ -19,65 +19,68 @@
 #ifndef PULSAR_AUTHENTICATION_H_
 #define PULSAR_AUTHENTICATION_H_
 
-#include <vector>
-#include <string>
-#include <map>
-#include <boost/shared_ptr.hpp>
 #include <pulsar/Result.h>
 #include <boost/make_shared.hpp>
+#include <boost/shared_ptr.hpp>
```

Comment:
I'm not sure whether we can do this, but I think we should not touch `#include` lines for now. Because reordering the lines sometimes causes compile errors. Although we should fix such errors, it would require extra changes that goes beyond formatting.

Reply:
I think for `boost/` includes there should be no problem. For `pulsar/` hopefully the same :). In general we should make sure that the headers are not depending on something they don't include by themselves.

Reply:
Right, let’s not touch them if we found the errors. One more thing we should make sure ideally is that headers or sources are not including they don’t depend on. There should be a tool that ensure the both, but I don’t remember the name.

Reply:
>  let’s not touch them if we found the errors

I suggest not to re order headers at all - cause I had at some point seen old boost headers (1.41) that were not self contained and caused issues on re ordering

To be specific these ones:-
```
+#include <boost/accumulators/framework/features.hpp>
 +
 +#include <boost/accumulators/accumulators.hpp>
 +#include <boost/accumulators/statistics.hpp>
 +#include <boost/accumulators/framework/accumulator_set.hpp>
 +#include <boost/accumulators/statistics/extended_p_square.hpp>
```

Reply:
> I suggest not to re order headers at all 

Personally I'm fine with it too. I've never been annoyed with unsorted `#include` lines. I mean, not-self-contained headers are annoying but I don't care the ordering so much if it is compilable.